crm_claim_rma/models/crm_claim.py

```diff
@@ -102,60 +99,106 @@ def _onchange_invoice(self):
     @api.onchange('warehouse_id', 'claim_type', 'date')
     def _onchange_invoice_warehouse_type_date(self):
         context = self.env.context
-        claim_line = self.env['claim.line']
         if not self.warehouse_id:
             self.warehouse_id = self._get_default_warehouse()
-        claim_type = self.claim_type
-        claim_date = self.date
-        warehouse = self.warehouse_id
-        company = self.company_id
         create_lines = context.get('create_lines')
 
-        def warranty_values(invoice, product):
-            values = {}
-            try:
-                warranty = claim_line._warranty_limit_values(
-                    invoice, claim_type, product, claim_date)
-            except (InvoiceNoDate, ProductNoSupplier):
-                # we don't mind at this point if the warranty can't be
-                # computed and we don't want to block the user
-                values.update({'guarantee_limit': False, 'warning': False})
-            else:
-                values.update(warranty)
-
-            warranty_address = claim_line._warranty_return_address_values(
-                product, company, warehouse)
-            values.update(warranty_address)
-            return values
-
         if create_lines:  # happens when the invoice is changed
-            claim_lines = []
-            invoices_lines = self.invoice_id.invoice_line_ids.filtered(
-                lambda line: line.product_id.type in ('consu', 'product')
-            )
-            for invoice_line in invoices_lines:
-                location_dest = claim_line.get_destination_location(
-                    invoice_line.product_id, warehouse)
+            self._create_claim_lines()
+
+        if self.invoice_id:
+            self.delivery_address_id = self.invoice_id.partner_id.id
+
+    @api.model
+    def __check_packages_enabled(self):
+        group = self.env.ref(
+            "stock.group_tracking_lot",
+            raise_if_not_found=False,
+        )
+        employee_group = self.env.ref("base.group_user")
+        # if package managing is not activated or group deleted
+        if (not group or
+                group.id not in employee_group.implied_ids.ids or
+                not group.users):
+            return False
+        return True
```

Comment:
why not use "self.user_has_groups('stock.group_tracking_lot')" ?

Reply:
Because user can be explicitly deleted from the implied group. And the correct packaging should be resolved either way.

Reply:
Make sense to check the option on the employee group.

Reply:
This module depends of `stock` if the group is deleted then a `-u stock` will fix the bad practice deleting groups.